apple/Handlers/RNPinchHandler.m

```diff
@@ -67,28 +76,23 @@ - (void)interactionsCancelled:(NSSet *)touches withEvent:(UIEvent *)event
 }
 
 #if TARGET_OS_OSX
-- (void)touchesBeganWithEvent:(NSEvent *)event
-{
-  [super touchesBeganWithEvent:event];
-  [self interactionsBegan:[NSSet setWithObject:event] withEvent:event];
-}
-
-- (void)touchesMovedWithEvent:(NSEvent *)event
-{
-  [super touchesMovedWithEvent:event];
-  [self interactionsMoved:[NSSet setWithObject:event] withEvent:event];
-}
-
-- (void)touchesEndedWithEvent:(NSEvent *)event
-{
-  [super touchesEndedWithEvent:event];
-  [self interactionsEnded:[NSSet setWithObject:event] withEvent:event];
-}
+- (void)magnifyWithEvent:(NSEvent *)event
+{
+  [super magnifyWithEvent:event];
+
+  if (self.state == NSGestureRecognizerStateBegan) {
+    [self interactionsBegan:[NSSet setWithObject:event] withEvent:event];
+  } else if (self.state == NSGestureRecognizerStateChanged) {
+    [self interactionsMoved:[NSSet setWithObject:event] withEvent:event];
+  } else if (self.state == NSGestureRecognizerStateEnded) {
+    [self interactionsEnded:[NSSet setWithObject:event] withEvent:event];
+  } else if (self.state == NSGestureRecognizerStateCancelled) {
+    [self interactionsCancelled:[NSSet setWithObject:event] withEvent:event];
+  }
 
-- (void)touchesCancelledWithEvent:(NSEvent *)event
-{
-  [super touchesCancelledWithEvent:event];
-  [self interactionsCancelled:[NSSet setWithObject:event] withEvent:event];
+  _velocity = (self.magnification - prevMagnification) / (event.timestamp - prevTime);
```

Comment:
To double-check, what's the unit of `event.timestamp`? On iOS the unit of timer is device-dependant (https://github.com/software-mansion/react-native-gesture-handler/blob/db51032fcd0f81449736564e2f3fd055f29a28a4/apple/Handlers/RNLongPressHandler.m#L117).

Reply:
Then the time delta should be multiplied by `1000` (both here and in the rotation handler), on Android the time unit we're using is milliseconds.